### ISSUE TITLE
Single node training

### ISSUE THREAD

**Issue by sachit-menon** [Visual-Agent/DeepEyes#14]
Hi, thanks for this great work! I noticed you mentioned recommendations for the smallest compute settings still requiring multiple nodes. Most academic groups don’t have the capacity for multinode training, so I was wondering what would work or what wouldn’t with e.g., 8 80GB A100s? How long did training take with the recommended settings, and how long might it take in the single node setting? What else would need to change to get the method working in that setting? Thanks!

**Comment by Alisaleelee** [comment]
> Yes. It is possible to train using a single node (1x8 GPUs) by setting all the `micro_batch_size_per_gpu` to 1 in the training scripts: https://github.com/Visual-Agent/DeepEyes/blob/main/examples/agent/final_merged_v1v8_thinklite.sh
> 
> actor_rollout_ref.actor.ppo_micro_batch_size_per_gpu=1
> actor_rollout_ref.rollout.log_prob_micro_batch_size_per_gpu=1
> actor_rollout_ref.ref.log_prob_micro_batch_size_per_gpu=1
> Here is the performance result on our machine:
> 
> * timing_s/gen:802.057 - timing_s/old_log_prob:293.286 - timing_s/adv:184.065 - timing_s/update_actor:860.063 - timing_s/step:2142.433 - timing_per_token_ms/gen:0.986 - timing_per_token_ms/update_actor:0.136 - timing_per_token_ms/adv:0.029 - perf/total_num_tokens:6426750.000 - perf/time_per_step:2142.433 - perf/throughput:374.968


If I’m training on a single node with 8 GPUs, does vLLM serve need to be run on a separate node? I only have 8 GPUs (on a single node), so what would be a reasonable way to allocate resources?
如果在单节点，8卡上训练的话，vllm serve需要在另外一个节点运行吗？我只有8张卡（单个节点），请问该怎么划分比较合理呢

**Comment by wcpcp** [comment]
> > Yes. It is possible to train using a single node (1x8 GPUs) by setting all the `micro_batch_size_per_gpu` to 1 in the training scripts: https://github.com/Visual-Agent/DeepEyes/blob/main/examples/agent/final_merged_v1v8_thinklite.sh
> > actor_rollout_ref.actor.ppo_micro_batch_size_per_gpu=1
> > actor_rollout_ref.rollout.log_prob_micro_batch_size_per_gpu=1
> > actor_rollout_ref.ref.log_prob_micro_batch_size_per_gpu=1
> > Here is the performance result on our machine:
> > 
> > * timing_s/gen:802.057 - timing_s/old_log_prob:293.286 - timing_s/adv:184.065 - timing_s/update_actor:860.063 - timing_s/step:2142.433 - timing_per_token_ms/gen:0.986 - timing_per_token_ms/update_actor:0.136 - timing_per_token_ms/adv:0.029 - perf/total_num_tokens:6426750.000 - perf/time_per_step:2142.433 - perf/throughput:374.968
> 
> If I’m training on a single node with 8 GPUs, does vLLM serve need to be run on a separate node? I only have 8 GPUs (on a single node), so what would be a reasonable way to allocate resources? 如果在单节点，8卡上训练的话，vllm serve需要在另外一个节点运行吗？我只有8张卡（单个节点），请问该怎么划分比较合理呢

How did you allocate and solve this problem？

**Comment by FloSophorae** [comment]
> > > Yes. It is possible to train using a single node (1x8 GPUs) by setting all the `micro_batch_size_per_gpu` to 1 in the training scripts: https://github.com/Visual-Agent/DeepEyes/blob/main/examples/agent/final_merged_v1v8_thinklite.sh
> > > actor_rollout_ref.actor.ppo_micro_batch_size_per_gpu=1
> > > actor_rollout_ref.rollout.log_prob_micro_batch_size_per_gpu=1
> > > actor_rollout_ref.ref.log_prob_micro_batch_size_per_gpu=1
> > > Here is the performance result on our machine:
> > > 
> > > * timing_s/gen:802.057 - timing_s/old_log_prob:293.286 - timing_s/adv:184.065 - timing_s/update_actor:860.063 - timing_s/step:2142.433 - timing_per_token_ms/gen:0.986 - timing_per_token_ms/update_actor:0.136 - timing_per_token_ms/adv:0.029 - perf/total_num_tokens:6426750.000 - perf/time_per_step:2142.433 - perf/throughput:374.968
> > 
> > 
> > If I’m training on a single node with 8 GPUs, does vLLM serve need to be run on a separate node? I only have 8 GPUs (on a single node), so what would be a reasonable way to allocate resources? 如果在单节点，8卡上训练的话，vllm serve需要在另外一个节点运行吗？我只有8张卡（单个节点），请问该怎么划分比较合理呢
> 
> How did you allocate and solve this problem？

同问

**Comment by wcpcp** [comment]
> > > > Yes. It is possible to train using a single node (1x8 GPUs) by setting all the `micro_batch_size_per_gpu` to 1 in the training scripts: https://github.com/Visual-Agent/DeepEyes/blob/main/examples/agent/final_merged_v1v8_thinklite.sh
> > > > actor_rollout_ref.actor.ppo_micro_batch_size_per_gpu=1
> > > > actor_rollout_ref.rollout.log_prob_micro_batch_size_per_gpu=1
> > > > actor_rollout_ref.ref.log_prob_micro_batch_size_per_gpu=1
> > > > Here is the performance result on our machine:
> > > > 
> > > > * timing_s/gen:802.057 - timing_s/old_log_prob:293.286 - timing_s/adv:184.065 - timing_s/update_actor:860.063 - timing_s/step:2142.433 - timing_per_token_ms/gen:0.986 - timing_per_token_ms/update_actor:0.136 - timing_per_token_ms/adv:0.029 - perf/total_num_tokens:6426750.000 - perf/time_per_step:2142.433 - perf/throughput:374.968
> > > 
> > > 
> > > If I’m training on a single node with 8 GPUs, does vLLM serve need to be run on a separate node? I only have 8 GPUs (on a single node), so what would be a reasonable way to allocate resources? 如果在单节点，8卡上训练的话，vllm serve需要在另外一个节点运行吗？我只有8张卡（单个节点），请问该怎么划分比较合理呢
> > 
> > 
> > How did you allocate and solve this problem？
> 
> 同问

i use the api of QWen instead of using VLLM inference server locally

**Comment by RohollahHS** [comment]
> Thank you! Am I reading it correctly to interpret it will take `2142.433*80/(60*60) = 47.6` hours or roughly two days with a single-node machine? Also, is this timing from an H100 node? Finally, are no changes needed to e.g., the learning rate for similar results in this regime?

Hi. Did you try to train DeepEyes on 8 GPUs? How long it took?
